### DISCUSSION TITLE
identify user-agent on incrementally static generated pages in Next Js 10

### DISCUSSION THREAD

**Discussion by Sawan1994** [vercel/next.js#21860] [Help]
I am using Contentful as headless CMS to generate article pages on my web app. I have used Next's iSSG. 

The header and footer in the code base are different for mobile and desktop. I need a way to identify whether user is calling app from mobile or desktop and serve appropriate page with correct header and footer.

NOTE: The page has to be SEO friendly so the links in the header and footer should be available in the page source on initial load.

**Comment by leerob** [comment]
You don't have access to the request with ISR, because it's happening in the background. If you need dynamic data _at request time_, then SSR is a better choice. 

However, if you're just trying to change the header/footer, can't you use media queries with CSS?

**Reply by iou90** [reply]
Put the entire app within an iframe on desktop.
